src/content/docs/en/guides/deploy/gitlab.mdx

```diff
@@ -6,39 +6,90 @@ i18nReady: true
 ---
 import { Steps } from '@astrojs/starlight/components';
 
-You can use [GitLab Pages](https://pages.gitlab.io/) to host an Astro site for your [GitLab](https://about.gitlab.com/) projects, groups, or user account.
+You can use [GitLab Pages](https://docs.gitlab.com/ee/user/project/pages/) to host an Astro site for your [GitLab](https://about.gitlab.com/) projects, groups, or user account.
 
 :::tip[Looking for an example?]
 Check out [the official GitLab Pages Astro example project](https://gitlab.com/pages/astro)!
 :::
 
 ## How to deploy
 
+You can deploy an Astro site to GitLab Pages by using GitLab CI/CD to automatically build and deploy your site. To do this, your source code must be hosted on GitLab and you need to make the following changes to your Astro project:
+
 <Steps>
-1. Set the correct `site` in `astro.config.mjs`.
 
-2. Rename the `public/` directory to `static`.
+1. Set up [`site`](/en/reference/configuration-reference/#site) and [`base`](/en/reference/configuration-reference/#base) options in `astro.config.mjs`.
+
+    ```js title="astro.config.mjs" ins={4-5}
+    import { defineConfig } from 'astro/config';
+
+    export default defineConfig({
+      site: 'https://<username>.gitlab.io',
+      base: '/<my-repo>',
+      outDir: 'public',
+      publicDir: 'static',
+    });
+    ```
+
+    `site`
+
+    The value for `site` must be one of the following:
+
+    - The following URL based on your username: `https://<username>.gitlab.io`
+    - The following URL based on your group name: `https://<groupname>.gitlab.io`
+    - Your custom domain if you have it configured in your Gitlab project’s settings: `https://example.com`
+
+    For GitLab self-managed instances, replace `gitlab.io` with your instance’s Pages domain.
+
+    `base`
+
+    A value for `base` may be required so that Astro will treat your repository name (e.g. `/my-repo`) as the root of your website.
+
+    :::note
+      Don't set a `base` parameter if your page is served from the root folder.
+    :::
+
+    The value for `base` should be your repository’s name starting with a forward slash, for example `/my-blog`. This is so that Astro understands your website's root is `/my-repo`, rather than the default `/`.
```

Comment:
Again, this is true for GitHub specifically, but is not something needed for every deploy host. Have you confirmed that GitLab works the same way?

Reply:
I also confirmed this works the same way.

Other than the GitLab Pages docs, there is also this blog post from GitLab:
[How to publish your Astro Site with GitLab Pages](https://about.gitlab.com/blog/2022/10/24/publishing-an-astro-site-with-pages/)

> Depending on your project configuration, GitLab Pages will deploy your site at a URL that follows the format simlar to https://<user-or-group>.gitlab. io/<project-name>. If you want to use the default URL, you need to adjust Astro to the fact that the site is not mounted at the root path, otherwise it may not load static assets (such as the CSS files) correctly.